WooCommerce/src/main/kotlin/com/woocommerce/android/model/GetLocations.kt

```diff
@@ -0,0 +1,22 @@
+package com.woocommerce.android.model
+
+import com.woocommerce.android.ui.orders.details.editing.address.LocationCode
+import org.wordpress.android.fluxc.store.WCDataStore
+import javax.inject.Inject
+
+class GetLocations @Inject constructor(private val locationStore: WCDataStore) {
+    operator fun invoke(countryCode: LocationCode, stateCode: LocationCode): Pair<Location, AmbiguousLocation> {
```

Comment:
```suggestion
    operator suspend fun invoke(countryCode: LocationCode, stateCode: LocationCode): Pair<Location, AmbiguousLocation> {
```

Ideally, this should cover threading as currently here, and other places, we perform database operations on main thread. I'll deal with that when I'll migrate `WCLocationModel` to Room.

Reply:
I agree about being it suspend function should be better, but I'm wondering how this would play with the current usages, since `toAppModel` is being used in multiple places, and it was always assumed that it's a fast non-blocking function 🤔.

Reply:
I believe that if we'll start to use `*Repository* class in more places, it won't be such issue as getting value and mapping will be called via the same, `suspend` function.

I agree that `toAppModel` is used in multiple places, but usually it's on database entities in presentation layers, sometimes including `json` parsing (like in case of `Order`). I believe we have to start treating `toAppModel` as not as lightweight functions and not using them in presentaion.